Sources/TuistKit/Services/InitService.swift

```diff
@@ -1,266 +1,207 @@
-                attributesDictionary[name] = .string(option)
-            case let .optional(name, default: defaultValue):
-                guard let unwrappedOption = optionalTemplateOptions[name],
-                      let option = unwrappedOption
-                else {
-                    attributesDictionary[name] = defaultValue
-                    return
-                }
-                attributesDictionary[name] = .string(option)
-            }
-        }
+        return nil
     }
 
-    /// Finds template directory
-    /// - Parameters:
-    ///     - templateDirectories: Paths of available templates
-    ///     - template: Name of template
-    /// - Returns: `AbsolutePath` of template directory
-    private func templateDirectory(templateDirectories: [AbsolutePath], template: String) throws -> AbsolutePath {
-        guard let templateDirectory = templateDirectories.first(where: { $0.basename == template })
-        else { throw InitServiceError.templateNotFound(template) }
-        return templateDirectory
-    }
-
-    /// Returns name to use. If `name` is nil, returns a directory name executed `init` command.
-    private func name(_ name: String?, path: AbsolutePath) throws -> String {
-        if let name {
-            return name
-        } else if let directoryName = path.components.last {
-            return directoryName
-        } else {
-            throw InitServiceError.ungettableProjectName(AbsolutePath.current)
-        }
-    }
-
-    private func path(_ path: String?) throws -> AbsolutePath {
-        if let path {
-            return try AbsolutePath(validating: path, relativeTo: FileHandler.shared.currentPath)
-        } else {
-            return FileHandler.shared.currentPath
-        }
+    private func nameOfXcodeProjectOrWorkspace(in directory: AbsolutePath) async throws -> StartPromptingWorkflowType {
+        let xcodeProjectsAndWorkspaces = try await findXcodeProjectsAndWorkspaces(in: directory)
+        return prompter
+            .promptWorkflowType(
+                xcodeProjectOrWorkspace: xcodeProjectsAndWorkspaces
+                    .first(where: \.isWorkspace) ?? xcodeProjectsAndWorkspaces.first(where: \.isProject)
+            )
     }
 
-    private func platform(_ platform: String?) throws -> Platform {
-        if let platformString = platform {
-            if let platform = Platform(rawValue: platformString) {
-                return platform
-            } else {
-                throw InitServiceError.invalidValue(argument: "platform", error: "Platform should be either ios, tvos, or macos")
-            }
-        } else {
-            return .iOS
-        }
+    private func findXcodeProjectsAndWorkspaces(in directory: AbsolutePath) async throws -> Set<XcodeProjectOrWorkspace> {
+        var paths = Set(
+            try await fileSystem.glob(directory: directory, include: ["**/*.xcworkspace"]).collect()
+                .map(XcodeProjectOrWorkspace.workspace)
```

Comment:
This goes unnecessarily deep – if you end up running such such a command in the root of your file hierarchy, we'd end up globbing your whole file system which can easily take 10s of minutes. Let's glob for a project only in the current directory (or limit globbing to two, three levels deep)

Reply:
How can I control the depth of the globbing? Is it `*/*/*.xcworkspace`?